### PR TITLE
Allow to execute specific function

### LINKED ISSUE #190: [STE-106] Support execution exported functions
We need the ability to execute guest exported functions. The current workaround to this is to have a "function" router in the _start function, which is very annoying. 

<sub>[STE-106](https://linear.app/stealthrocket/issue/STE-106/support-execution-exported-functions)</sub>

### PR DESCRIPTION
**Pull request by Pryz**
Fixes #190 

Add the ability to define which function to execute within the WASM module. The function needs to be exported so this will not work with Go 1.21 but will work for Tinygo, Rust and friends.

### CODE REVIEW

File: internal/timecraft/module.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ type ModuleSpec struct {
 	// Path is the path of the WebAssembly module.
 	Path string
 
+	// Name of the exported function to call in the WebAssembly module.
+	// If empty, the _start function will be executed.
+	Function string
```
Comment (achille-roussel) [inline internal/timecraft/module.go:20]: At first sight it seems the use cases for executing anything other than _start (which has no args and no return value) might be very limited, do we need a way to pass arguments and retrieve the results?

Comment (achille-roussel) [reply internal/timecraft/module.go:20]: Did you look at how extism does it? It seems they've got a somewhat cross-language mechanism for invoking exported functions with rich parameters

Comment (Pryz) [reply internal/timecraft/module.go:20]: They use a C binding. Similar to wit-binding or spin.

Comment (achille-roussel) [reply internal/timecraft/module.go:20]: But how do they solve for the calling convention? The data needs to be passed between the host and the guest, there's more than C bindings to achieve that.

Comment (Pryz) [reply internal/timecraft/module.go:20]: Will be part of another PR.

**Review by achille-roussel — APPROVED** [review]

**Review by achille-roussel — APPROVED** [review]

**Pull request merged**